nixos/modules/services/hardware/zcfan.nix

```
@@ -0,0 +1,59 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.zcfan;
+in
+{
+  options = {
+    services.zcfan = {
+      enable = lib.mkEnableOption "zero-configuration fan daemon for ThinkPads (zcfan)";
+
+      settings = lib.mkOption {
+        type = lib.types.lines;
+        default = "";
+        description = ''
+          Optional lines added verbatim to the configuration file.
+          Before changing this, read the {manpage}`zcfan(1)`
+          and project's readme at
+          https://github.com/cdown/zcfan/tree/master?tab=readme-ov-file#usage
+        '';
+      };
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !config.services.thinkfan.enable;
+        message = ''
+          You have set services.zcfan.enable = true;
+          which conflicts with services.thinkfan.enable = true;
+        '';
+      }
+    ];
+
+    environment.systemPackages = [ pkgs.zcfan ];
+    systemd.packages = [ pkgs.zcfan ];
+
+    environment.etc."zcfan.conf".text = cfg.settings;
+
+    systemd.services = {
+      zcfan = {
+        serviceConfig = {
+          Restart = "on-failure";
+          RestartSec = "30s";
```

Could you check whether the service still works with granular access?

```suggestion
          RestartSec = "30s";
          
          PrivateDevices = true;
          ProtectKernelTunables = true;
          ReadWritePaths = [
            "/proc/acpi/ibm/fan"
            "/sys/class/hwmon"
          ];
```

Ref https://github.com/cdown/zcfan/blob/42b149b8c6bf1823b19e7c026d82c0f1c9863e52/zcfan.c#L13-L14